src/SmartTransactionsController.ts

```diff
@@ -47,165 +48,237 @@ import {
+const DEFAULT_CLIENT_ID = 'default';
 const ETH_QUERY_ERROR_MSG =
   '`ethQuery` is not defined on SmartTransactionsController';
 
-export type SmartTransactionsControllerConfig = BaseConfig & {
-  interval: number;
-  clientId: string;
-  chainId: Hex;
-  supportedChainIds: Hex[];
+/**
+ * The name of the {@link SmartTransactionsController}
+ */
+const controllerName = 'SmartTransactionsController';
+
+const controllerMetadata = {
+  smartTransactionsState: {
+    persist: false,
+    anonymous: true,
+  },
 };
 
 type FeeEstimates = {
-  approvalTxFees: IndividualTxFees | undefined;
-  tradeTxFees: IndividualTxFees | undefined;
+  approvalTxFees: IndividualTxFees | null;
+  tradeTxFees: IndividualTxFees | null;
 };
 
-export type SmartTransactionsControllerState = BaseState & {
+export type SmartTransactionsControllerState = {
   smartTransactionsState: {
     smartTransactions: Record<Hex, SmartTransaction[]>;
-    userOptIn: boolean | undefined;
-    userOptInV2: boolean | undefined;
-    liveness: boolean | undefined;
+    userOptIn: boolean | null;
+    userOptInV2: boolean | null;
+    liveness: boolean | null;
     fees: FeeEstimates;
     feesByChainId: Record<Hex, FeeEstimates>;
     livenessByChainId: Record<Hex, boolean>;
   };
 };
 
-export default class SmartTransactionsController extends StaticIntervalPollingControllerV1<
-  SmartTransactionsControllerConfig,
-  SmartTransactionsControllerState
+/**
+ * Get the default {@link SmartTransactionsController} state.
+ *
+ * @returns The default {@link SmartTransactionsController} state.
+ */
+export function getDefaultSmartTransactionsControllerState(): SmartTransactionsControllerState {
+  return {
+    smartTransactionsState: {
+      smartTransactions: {},
+      userOptIn: null,
+      userOptInV2: null,
+      fees: {
+        approvalTxFees: null,
+        tradeTxFees: null,
+      },
+      liveness: true,
+      livenessByChainId: {
+        [ChainId.mainnet]: true,
+        [ChainId.sepolia]: true,
+      },
+      feesByChainId: {
+        [ChainId.mainnet]: {
+          approvalTxFees: null,
+          tradeTxFees: null,
+        },
+        [ChainId.sepolia]: {
+          approvalTxFees: null,
+          tradeTxFees: null,
+        },
+      },
+    },
+  };
+}
+
+export type SmartTransactionsControllerGetStateAction =
+  ControllerGetStateAction<
+    typeof controllerName,
+    SmartTransactionsControllerState
+  >;
+
+/**
+ * The actions that can be performed using the {@link SmartTransactionsController}.
+ */
+export type SmartTransactionsControllerActions =
+  SmartTransactionsControllerGetStateAction;
+
+export type AllowedActions = NetworkControllerGetNetworkClientByIdAction;
+
+export type SmartTransactionsControllerStateChangeEvent =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    SmartTransactionsControllerState
+  >;
+
+export type SmartTransactionsControllerSmartTransactionEvent = {
+  type: 'SmartTransactionsController:smartTransaction';
```

Comment:
nit: Do you think we can make this event name more meaningful by describing what happened with the smart transaction? Perhaps something like `SmartTransactionsController:smartTransactionCreated`?